### PR TITLE
Legend items should be sorted the same way as series do

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
Legend items should be sorted the same way as series do. 
Series line color is white to be consistent with other charts

* Functional Testing 
  * [x] Lit 2 Phase 1 compliant changes
  * [x] Unit tests
* Security, devops, perf: N/A
* UX and docs: N/A

#### New <= Prev

image

### CODE REVIEW

File: model/data.js [hunk 1]
```diff
@@ -18,6 +19,9 @@ export class Data {
 	}
 
 	getDataSet(name) {
+		if (!this._dataSets.has(name)) {
+			this._dataSets.set(name, this._dataSetFactories[name](this));
+		}
```
Comment (Vitalii-Misechko) [inline model/data.js:24]: Dataset lazy initialization

**Comment by bemailloux** [comment]
Interesting - so the previous version had the incorrect legend for each tool usage trend?

**Comment by Vitalii-Misechko** [comment]
> Interesting - so the previous version had the incorrect legend for each tool usage trend?

That's correct

**Review by bemailloux — COMMENTED** [review]
LGTM!

**Review by bemailloux — APPROVED** [review]

File: model/dataSets/toolUsageTrendByRoleDataSet.js [hunk 2]
```diff
@@ -33,13 +33,16 @@ export class ToolUsageTrendByRoleDataSet extends DataSet {
...
-		});
+		return sortBy(
+			Object.entries(toolsUsageDataMap).map(([roleId, usageData]) => {
+				return {
+					id: roleId,
+					name: data.lmsData?.rolesInfoMap[roleId] ?? roleId,
+					usageData
+				};
+			}),
+			'name'
```
[the remark this answers is not in the document]
Comment (johngwilkinson) [reply model/dataSets/toolUsageTrendByRoleDataSet.js:48]: Maybe out of scope: I believe we should sort by role name then role id. I'm adding a compare function to do that for org units in an upcoming PR. We should probably also localize the role labels including the id as for org unit names - not in this PR though

File: model/dataSets.js [hunk 3]
```diff
@@ -41,10 +40,10 @@ export const dataSetFactories = {
 		(data) => new ToolUsageComparisonDataSet(data.getResultSet(resultSets.TOOL_USAGE_TREND)),
 
 	[dataSets.TOOL_USAGE_TREND_BY_TOOL_LEGEND]:
-		(data) => new ToolUsageLegendDataSet(data.getResultSet(resultSets.TOOL_USAGE_TREND)),
+		(data) => new ToolUsageTrendLegendDataSet(data.getDataSet(dataSets.TOOL_USAGE_TREND_BY_TOOL)),
```
Comment (johngwilkinson) [inline model/dataSets.js:43]: oh, neat trick!

**Review by johngwilkinson — APPROVED** [review]

File: model/dataSets/toolUsageTrendByRoleDataSet.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Vitalii-Misechko) [reply model/dataSets/toolUsageTrendByRoleDataSet.js]: Added sorting by `name` and `id` across Tool Usage Trend and Tool Usage Comparison

**Review by bemailloux — APPROVED** [review]

**Review by johngwilkinson — APPROVED** [review]

**Pull request merged**